Ticketing/Ticketing.BAL/Services/CartService.cs

```diff
@@ -69,17 +69,27 @@ public async Task<int> BookSeatToCartAsync(Guid cartId)
 
             var shoppingCartItems = shoppingCarts.Where(c => c.CartId == cartId).ToList();
 
+            if (!shoppingCartItems.Any())
+            {
+                return 0;
+            }
+
             var shoppingCartSeats = shoppingCartItems.Select(sh => sh.SeatId).ToList();
 
             var allSeats = await _repositorySeat.GetAllAsync();
 
             var seats = allSeats.Where(s => shoppingCartSeats.Contains(s.Id));
 
+            if (!seats.Any())
+            {
+                return 0;
```

Comment:
1) Magic number
2) It is better to avoid any error codes like behavior and substitute it OR with exception OR with null in your case

Reply:
done